firebase-common/src/commonMain/kotlin/dev/gitlive/firebase/serializers.kt

```diff
@@ -104,3 +104,30 @@ class FirebaseListSerializer : KSerializer<Iterable<Any?>> {
     }
 }
 
+/**
+ * A special case of serializer for values natively supported by Firebase and
+ * don't require an additional encoding/decoding.
+ */
+abstract class SpecialValueSerializer<T>(
```

Comment:
in this PR it's used for Realtime Database `ServerValue` and Firestore `FieldValue` and `Timestamp`. _Firestore also supports storing `GeoPoints` and `DocumentReferences` natively, but it's out of scope of this PR_